### DISCUSSION TITLE
draco-rc5: CMake not building app tables

### DISCUSSION THREAD

**Discussion by mdunn47** [nasa/cFS#815] [Q&A] [answered]
While testing with the draco-rc5 cFS branch, I ran into some problems creating the shared object files for the tables in the reusable apps (FM, SC, DS, etc). The apps currently bundled with draco-rc5 cFS (sample_app, to_lab, etc.) correctly find and compile their associated tables. When I attempted to add the draco-rc5 branch of File Manager to the app it properly built but the fm_monitor table is ignored in CMake.

When I edit CMakeLists for the FM app to add the cfe table before the FM app module is created
`add_cfe_tables(fm fsw/tables/fm_monitor.c)`
`add_cfe_app(fm ${APP_SRC_FILES} ${FM_OPTION_SRC_FILES})`

the table is found but gives this warning:

`NOTE: "fm" passed to add_cfe_tables is not a previously-defined application target`
`-- Using file: cfs/apps/FM/fsw/tables/fm_monitor.c for fm_monitor.c`
`-- Using cfs/apps/FM/fsw/tables/fm_monitor.c as table definition for fm_monitor on cpu1`

Has anyone experienced this on draco or equuleus?

**Comment by RR-HSz** [comment, marked as answer]
Hello!
I had a similar problem. The solution was to rename the application's folder to lowercase. It should be:
cfs/apps/fm/






[cid:a7e149f0-977f-498b-988f-4f1ec61d0b1c]

Szilárd HELF

Embedded Software Engineer

Design Engineering

Embedded Software Engineering

| DEWE |



Remred Ltd.

29-33 Konkoly-Thege Miklos Str.
H-1121 Budapest, HUNGARY

+36 (20) 216 7626

***@***.******@***.***>

remred.space<https://remred.space/>


________________________________
Feladó: mdunn47 ***@***.***>
Elküldve: 2024. december 20., péntek 16:19
Címzett: nasa/cFS ***@***.***>
Másolatot kap: Subscribed ***@***.***>
Tárgy: [nasa/cFS] draco-rc5: CMake not building app tables (Discussion #815)


While testing with the draco-rc5 cFS branch, I ran into some problems creating the shared object files for the tables in the reusable apps (FM, SC, DS, etc). The apps currently bundled with draco-rc5 cFS (sample_app, to_lab, etc.) correctly find and compile their associated tables. When I attempted to add the draco-rc5 branch of File Manager to the app it properly built but the fm_monitor table is ignored in CMake.

When I edit CMakeLists for the FM app to add the cfe table before the FM app module is created
add_cfe_tables(fm fsw/tables/fm_monitor.c)
add_cfe_app(fm ${APP_SRC_FILES} ${FM_OPTION_SRC_FILES})

the table is found but gives this warning:

NOTE: "fm" passed to add_cfe_tables is not a previously-defined application target
-- Using file: cfs/apps/FM/fsw/tables/fm_monitor.c for fm_monitor.c
-- Using cfs/apps/FM/fsw/tables/fm_monitor.c as table definition for fm_monitor on cpu1

Has anyone experienced this on draco or equuleus?
